### PR TITLE
Release relayer

### PR DESCRIPTION
**Pull request by yrong**
Resolves: SNO-558

Also include some fix for a non-local setup(i.e. use non-local/public endpoints for substrate chain nodes).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -3,36 +3,53 @@ name: release
...
+        uses: docker/login-action@v2.2.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
+        run: |
+          docker buildx build \
+          --platform linux/amd64 \
+          -f relayer/Dockerfile \
+          -t "ronyang/snowbridge-relay:${{ github.event.inputs.tag || github.sha }}" \
```
Comment (yrong) [inline .github/workflows/release.yml:53]: @vgeddes Except for the binary format I would suggest we add docker image as another option for running relay service. This is also the recomended way for running most of the other components in our stack.

Problem is that I don't have authority to config the repo so use my own account here temporarily. We need to register an account in docker hub as orgnization and add secrets here so to push image.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -3,36 +3,53 @@ name: release
 on:
   push:
     tags:
-      - v0.*
-      - v1.*
+      - "v*"
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: tag (ex. v1.0) to publish
```
Comment (claravanstaden) [inline .github/workflows/release.yml:10]:
```suggestion
        description: tag (e.g. v1.0) to publish
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (vgeddes) [reply .github/workflows/release.yml:53]: Actually in the past we've used GitHub Packages as a docker repository: https://github.com/Snowfork/snowbridge/pkgs/container/snowbridge-relay

Can you try and upload the image there to test?

Comment (yrong) [reply .github/workflows/release.yml:53]: Cool! Seems also require to config `GITHUB_TOKEN` for [Authenticating to the Container registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry), so is that already configured in this repo before?

Comment (yrong) [reply .github/workflows/release.yml:57]: Some revamp in https://github.com/Snowfork/snowbridge/pull/881/commits/81f3bf07519bdc432ed55168128edd747d131415 also add a [workflow_dispatch](https://github.com/Snowfork/snowbridge/blob/765709754952dcabd263ca3ece19af5fd3987a6c/.github/workflows/release.yml#L7-L11) for manually trigger the action for test.

But seems without first merging to main it's not easy to test the newly added action.

**Review by alistair-singh — APPROVED** [review]

**Pull request closed without merging**